pkg/controllers/workgenerator/controller_test.go

```diff
@@ -1736,18 +1730,31 @@ func TestUpdateBindingStatusWithRetry(t *testing.T) {
 			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(tt.resourceBinding), binding); err != nil {
 				t.Errorf("updateBindingStatusWithRetry() error = %v, wantErr %v", err, nil)
 			}
-
-			if tt.conflictCount > 0 {
-				latestRollout := tt.latestBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted))
-				rollout := tt.resourceBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted))
-				if diff := cmp.Diff(latestRollout, rollout, ignoreOption); diff != "" {
-					t.Errorf("updateBindingStatusWithRetry() ResourceBindingRolloutStarted Condition got = %v, want %v", rollout, latestRollout)
+			latestRollout := tt.latestBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted))
+			rollout := binding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted))
+			if diff := cmp.Diff(latestRollout, rollout, ignoreOption); diff != "" {
+				t.Errorf("updateBindingStatusWithRetry() ResourceBindingRolloutStarted Condition got = %v, want %v", rollout, latestRollout)
+			} else {
+				// A tolerance is used when comparing timestamps to account for precision differences
+				if !CompareTimeWithTolerance(rollout.LastTransitionTime, latestRollout.LastTransitionTime, 1*time.Second) {
+					t.Errorf("updateBindingStatusWithRetry() ResourceBindingRolloutStarted LastTransitionTime got = %v, want %v", rollout.LastTransitionTime, latestRollout.LastTransitionTime)
 				}
 			}
 		})
 	}
 }
 
+func CompareTimeWithTolerance(t1, t2 metav1.Time, tolerance time.Duration) bool {
```

Comment:
you can refer to this compare option, https://github.com/Azure/fleet/blob/main/pkg/controllers/clusterresourceplacement/placement_status_test.go#L47